glean-core/src/database/mod.rs

```diff
@@ -243,6 +284,17 @@ impl Database {
             return Ok(());
         }
 
+        if lifetime == Lifetime::Ping {
+            let ping_lifetime_data = &self.ping_lifetime_data;
```

Comment:
Both the reading and the writing parts for `Lifetime::Ping` and `Lifetime::Application` are kind of identical. Did you consider sharing the code in some utility functioN?

Reply:
Yes, I agree. I am looking into doing something like that before marking this as "done" :)

Reply:
I looked into this, and the code is similar but not exaclty the same specially because of the `if let`, I could figure out a way to this nicely. Let me know if you think of something :)

Reply:
I was talking to @badboy and he suggested filing a bug to look into this after this PR is done. What do you think?

Reply:
Fine by me!